service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java

```diff
@@ -234,6 +274,33 @@ public InetAddress getServerIPAddress() {
     return serverIPAddress;
   }
 
+  private void logAuditEvent(String cmd) throws HiveSQLException {
+    if (cmd == null) {
+      return;
+    }
+
+    final Formatter fmt = auditFormatter.get();
+    ((StringBuilder) fmt.out()).setLength(0);
+
+    String address = getIpAddress();
+    if (address == null) {
+      address = "unknown-ip-addr";
+    }
+    auditLog.info(fmt.format(AUDIT_FORMAT, getUserName(), address, cmd).toString());
+  }
+
+  public void startFunction(ThriftCliFunctions function, String extraLogInfo)
+      throws HiveSQLException {
+    logAuditEvent(function + extraLogInfo);
+  }
+
+  public void startFunction(ThriftCliFunctions function) throws HiveSQLException {
+    startFunction(function, "");
```

Comment:
In L296 you do an implicit string conversion plus a string concatenation, you can do the string conversion directly here, and avoid the string concatenation with an empty string ... not sure if this would be optimized out otherwise.